deploy_heroku.py

```diff
@@ -0,0 +1,54 @@
+import argparse
+import json
+import os
+import subprocess
+import urllib.request
+from typing import List
+
+
+def get_latest_heroku_release(app: str, api_key: str):
+    request = urllib.request.Request(f"https://api.heroku.com/apps/{app}/releases")
+    for key, value in (
+            ("Accept", "application/vnd.heroku+json; version=3"),
+            ("Authorization", f"Bearer {api_key}"),
+            ("Content-Type", "application/json"),
+            ("Range", "version ..; order=desc,max=10;"),
+    ):
+        request.add_header(key, value)
+
+    with urllib.request.urlopen(request) as response:
+        assert response.getcode() == 200, response.getcode()
+        return json.loads(response.read())[0]
```

Comment:
Warum hast du denn der dataclass weggetan? ich fand ich eigentlich ganz schön, weil Felder sind dadurch stark typisiert und aufrufe per parse=time gecheckt, also nicht stringly typed. Meine Lösung war nur die deklarierte Schlüssel zu übernehmen und dadurch ganz generisch.

Hat da irgendwas nicht geklappt?

Reply:
Ja, ich musste das leider rausnehmen. :(
Fande es auch anders schöner, aber das Problem ist, dass die Actions derzeit mit Python3.6.9 laufen. Dataclasses ist erst mit Python3.7 hinzugekommen.

Ich habe gelesen, dass die Anfang Dezember den Rollout starten und die Actions auf Ubuntu-20 als default setzen, damit hätten wir Python3.8 als Default und könnten das dann wieder umbauen. Die werden den Rollout rollierend machen, somit wird sich das ins neue Jahr ziehen.

Reply:
Ahso, es ist noch 3.6, dann verstehe ich. Dann hätte ich es auch ausgebaut. Ich frage mich ob wir nicht den patch direkt ins repository ablegen sollen, um es nicht zu vergessen es zu applien wenn denn 3.8 kommt. Ist zwar irgendwie alles in PR aufbewahren, aber wird nach 1 Jahr nicht trivial es rauszuziehen und man muss noch daran denken. Aber ein Datei stört sofort :)

Reply:
Ich mache grade noch einen Versuch.
Mir ist aufgefallen, dass die Action die Runtime aus der `deploy.yml` nimmt. Dort ist derzeit `ubuntu-latest` gesetzt, was derezeit Ubuntu18.04 ist (https://github.com/actions/virtual-environments#available-environments). Wie oben schon beschrieben, werden die latest auf Ubuntu20.04 setzen.

Wir können aber in ùnserer `deploy.yml` einfach `ubuntu-20.04` anstatt `ubuntu-latest` setzen, dann haben wir auch Python3.8.
Ich werde dann ein Ticket erstellen, dass wieder auf `ubuntu-latest` zu ändern, sobald die den Default geändert haben.

Reply:
Okay, das scheint zu klappen. :)

Reply:
❤️